packages/commerce-sdk-react/src/hooks/ShopperBaskets/mut.test.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {act} from '@testing-library/react'
+import {ShopperBasketsTypes, ShopperCustomersTypes} from 'commerce-sdk-isomorphic'
+import nock from 'nock'
+import {
+    assertUpdateQuery,
+    DEFAULT_TEST_CONFIG,
+    mockMutationEndpoints,
+    mockQueryEndpoint,
+    renderHookWithProviders
+} from '../../test-utils'
+import {useCustomerBaskets} from '../ShopperCustomers'
+import {ApiClients, Argument} from '../types'
+import {ShopperBasketsMutation, useShopperBasketsMutation} from './mutation'
+import * as queries from './query'
+
+jest.mock('../../auth/index.ts', () => {
+    return jest.fn().mockImplementation(() => ({
+        ready: jest.fn().mockResolvedValue({access_token: 'access_token'})
+    }))
+})
+
+type Client = ApiClients['shopperBaskets']
+type Basket = ShopperBasketsTypes.Basket
+type BasketsResult = ShopperCustomersTypes.BasketsResult
+
+/** Create an options object for Shopper Baskets endpoints, with `basketId` pre-filled. */
+const makeOptions = <Method extends Exclude<keyof Client, 'clientConfig'>>(
+    body: Argument<Client[Method]> extends {body: infer B} ? B : undefined,
+    parameters: Omit<Argument<Client[Method]>['parameters'], 'basketId'>
+): Argument<Client[Method]> => ({
+    body,
+    parameters: {basketId: BASKET_ID, ...parameters}
+})
+
+// --- getBasket constants --- //
+const basketsEndpoint = '/checkout/shopper-baskets/'
+const BASKET_ID = 'basket_id'
+const getBasketOptions = makeOptions<'getBasket'>(undefined, {})
+const oldBasket: Basket = {basketId: BASKET_ID, mockData: 'old basket'}
+const newBasket: Basket = {basketId: BASKET_ID, mockData: 'new basket'}
+// --- getCustomerBaskets constants --- //
+const customersEndpoint = '/customer/shopper-customers/'
+const CUSTOMER_ID = 'customer_id'
+// Can't use `makeOptions()` here because it's Shopper Customers, not Shopper Baskets
+const getCustomerBasketsOptions: Argument<ApiClients['shopperCustomers']['getCustomerBaskets']> = {
+    parameters: {
+        customerId: CUSTOMER_ID
+    }
+}
+const oldCustomerBaskets: BasketsResult = {
+    // We aren't implementing the full basket, so we assert to pretend we are
+    baskets: [{basketId: 'other_basket'}, oldBasket] as BasketsResult['baskets'],
```

Comment:
We need to. make sure that the basketId here is consistent with the basket response from mutation, if we re-use the oldBasket/newBasket data as a response for mutation, we have to make sure this basketId is 'BASKET_ID'. Otherwise, the mutation won't get updated since we check the id before we allow the mutation to update

Reply:
What do you mean here? The basket being used _is_ `oldBasket` (and, `newBasket` for `newCustomerBaskets`), so the ID (and full basket contents) are re-used. There are two different baskets in the `baskets` array to validate that the correct basket gets updated.